app/database/accessor/breeze_buddy/template.py

```
@@ -429,6 +431,78 @@ async def get_template_by_outbound_number_id(
+        query, values = check_template_usage_query(template_id)
+        result = await run_parameterized_query(query, values)
+
+        usage: Dict[str, int] = {}
+        if result:
+            for row in result:
+                usage[row["source"]] = row["reference_count"]
+
+        logger.info(f"Template {template_id} usage: {usage}")
+        return usage
+
+    except Exception as e:
```

Ruff (0.15.2)

[warning] Do not catch blind exception: `Exception`

(BLE001)